sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -513,16 +513,21 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
   /**
    * Gets a student application by applicationId.
-   * Student id can be provided for authorization purposes.
+   * Student id/ institution id can be provided for authorization purposes.
    * @param applicationId application id.
    * @param options object that should contain:
    * - `loadDynamicData` indicates if the dynamic data(JSONB) should be loaded.
    * - `studentId` student id.
+   * - `institutionId` institution id.
    * @returns student application.
    */
   async getApplicationById(
     applicationId: number,
-    options?: { loadDynamicData?: boolean; studentId?: number },
+    options?: {
+      loadDynamicData?: boolean;
+      studentId?: number;
+      institutionId?: number;
```

Comment:
@guru-aot  @andrewsignori-aot @ann-aot @andrepestana-aot  @sh16011993 

I was wondering that, with so many options in place which justifies the service as such but in this particular scenario where this service is required to be used with studentId and institutionId. Hence from the consumption perspective we need application details of a student who applied to a given institution and IMO we can introduce a service which is more confined to do this by re-using existing service. 

like this

```ts
async getApplicationByStudentAndInstitution(
    applicationId: number,
    studentId: number,
    institutionId: number,
    options?: { loadDynamicData?: boolean },
  ) {
    return this.getApplicationById(applicationId, {
      studentId,
      institutionId,
      loadDynamicData: options?.loadDynamicData,
    });
  }
```

Reply:
what is the benefit here @dheepak-aot? I think I am not getting it

Reply:
I am not sure if I am following. Right now the method is consumed 3 times where the only parameter that is required is the `applicationId`. The idea would be to have more methods where the parameters would be required?

Reply:
That is right @andrewsignori-aot. Not part of this PR. But we can discuss outside about this idea.
@ann-aot It was exactly what @andrewsignori-aot below your comment.